lib/galaxy/jobs/__init__.py

```diff
@@ -1743,7 +1743,14 @@ def _change_ownership( self, username, gid ):
         job = self.get_job()
         # FIXME: hardcoded path
         external_chown_script = self.get_destination_configuration("external_chown_script", None)
-        cmd = [ '/usr/bin/sudo', '-E', external_chown_script, self.working_directory, username, str( gid ) ]
+
+        cmd = [ '/usr/bin/sudo', '-E', ]
+        for v in ["PATH", "LD_LIBRARY_PATH", "PKG_CONFIG_PATH"]:
```

Comment:
For the PATH we also thought that it is sufficient to hard code the path to the python binary in the script. But then there is still the problem with the python shared object file which is not found. Maybe one of you has an idea? Maybe I could build a special statically linked python binary just for these scripts. Could be a better idea than adding a potential security hole to galaxy.